File: uhrwerk-common/src/main/java/io/qimia/uhrwerk/common/model/ConnectionModelBuilder.java
package io.qimia.uhrwerk.common.model;

import org.jetbrains.annotations.NotNull;

public class ConnectionModelBuilder extends StateModelBuilder<ConnectionModelBuilder> {

  Long id;
  String name;
  ConnectionType type;
  String path;
  String jdbcUrl;
  String jdbcDriver;
  String jdbcUser;
  String jdbcPass;
  String awsAccessKeyID;
  String awsSecretAccessKey;

  public ConnectionModelBuilder id(Long id) {
    this.id = id;
    return this;
  }

  public ConnectionModelBuilder name(@NotNull String name) {
    this.name = name;
    return this;
  }

  public ConnectionModelBuilder type(ConnectionType type) {
    this.type = type;
    return this;
  }

  public ConnectionModelBuilder path(String path) {
    this.path = path;
    return this;
  }

  public ConnectionModelBuilder jdbcUrl(String jdbcUrl) {
    this.jdbcUrl = jdbcUrl;
    return this;
  }

  public ConnectionModelBuilder jdbcDriver(String jdbcDriver) {
    this.jdbcDriver = jdbcDriver;
    return this;
  }

  public ConnectionModelBuilder jdbcUser(String jdbcUser) {
    this.jdbcUser = jdbcUser;
    return this;
  }

  public ConnectionModelBuilder jdbcPass(String jdbcPass) {
    this.jdbcPass = jdbcPass;
    return this;
  }

  public ConnectionModelBuilder awsAccessKeyID(String awsAccessKeyID) {
    this.awsAccessKeyID = awsAccessKeyID;
    return this;
  }

  public ConnectionModelBuilder awsSecretAccessKey(String awsSecretAccessKey) {
    this.awsSecretAccessKey = awsSecretAccessKey;
    return this;
  }

  public ConnectionModel build() {
    return new ConnectionModel(this);
  }

  @Override
  ConnectionModelBuilder getThis() {
    return this;
  }
}

File: uhrwerk-metastore/src/main/kotlin/io/qimia/uhrwerk/common/metastore/builders/ConnectionModelBuilder.kt
package io.qimia.uhrwerk.common.metastore.builders

import io.qimia.uhrwerk.common.metastore.model.ConnectionModel
import io.qimia.uhrwerk.common.metastore.model.ConnectionType

class ConnectionModelBuilder : StateModelBuilder<ConnectionModelBuilder>() {
    var id: Long? = null
    var name: String? = null
    var type: ConnectionType? = null
    var path: String? = null
    var jdbcUrl: String? = null
    var jdbcDriver: String? = null
    var jdbcUser: String? = null
    var jdbcPass: String? = null
    var awsAccessKeyID: String? = null
    var awsSecretAccessKey: String? = null
    var redshiftFormat: String? = null
    var redshiftAwsIamRole: String? = null
    var redshiftTempDir: String? = null

    fun id(id: Long?): ConnectionModelBuilder {
        this.id = id
        return this
    }

    fun name(name: String): ConnectionModelBuilder {
        this.name = name
        return this
    }

    fun type(type: ConnectionType?): ConnectionModelBuilder {
        this.type = type
        return this
    }

    fun path(path: String?): ConnectionModelBuilder {
        this.path = path
        return this
    }

    fun jdbcUrl(jdbcUrl: String?): ConnectionModelBuilder {
        this.jdbcUrl = jdbcUrl
        return this
    }

    fun jdbcDriver(jdbcDriver: String?): ConnectionModelBuilder {
        this.jdbcDriver = jdbcDriver
        return this
    }

    fun jdbcUser(jdbcUser: String?): ConnectionModelBuilder {
        this.jdbcUser = jdbcUser
        return this
    }

    fun jdbcPass(jdbcPass: String?): ConnectionModelBuilder {
        this.jdbcPass = jdbcPass
        return this
    }

    fun awsAccessKeyID(awsAccessKeyID: String?): ConnectionModelBuilder {
        this.awsAccessKeyID = awsAccessKeyID
        return this
    }

    fun awsSecretAccessKey(awsSecretAccessKey: String?): ConnectionModelBuilder {
        this.awsSecretAccessKey = awsSecretAccessKey
        return this
    }

    fun redshiftFormat(redshiftFormat: String?): ConnectionModelBuilder {
        this.redshiftFormat = redshiftFormat
        return this
    }

    fun redshiftAwsIamRole(redshiftAwsIamRole: String?): ConnectionModelBuilder {
        this.redshiftAwsIamRole = redshiftAwsIamRole
        return this
    }

    fun redshiftTempDir(redshiftTempDir: String?): ConnectionModelBuilder {
        this.redshiftTempDir = redshiftTempDir
        return this
    }


    fun build(): ConnectionModel {
        return ConnectionModel(
            id,
            name,
            type,
            path,
            jdbcUrl,
            jdbcDriver,
            jdbcUser,
            jdbcPass,
            awsAccessKeyID,
            awsSecretAccessKey,
            redshiftFormat,
            redshiftAwsIamRole,
            redshiftTempDir
        )
    }

    override fun getThis() = this

}